### ISSUE TITLE
MSSQL LIMIT(top) causes Query input data to be put in wrong spot

### ISSUE THREAD

**Issue by starsoccer** [tgriesser/knex#1758]
I was trying to use knex to make a simply query but it seems to be switching up all of the variables in the query.

For example, My code is, 

console.log(knex('transactions').limit(1).where({ "transactionid": 'bcc021e6db2620680e773fd7c5d8726be8f74902728b80397f99fe8445abb403' }).update({ 'currentstatus': '-1' }).toString());

Yet it returns,

update top ('-1') [transactions] set [currentstatus] = 'bcc021e6db2620680e773fd7c5d8726be8f74902728b80397f99fe8445abb403' where [transactionid] = 1;

It seems to be taking what I set as the currentstatus, and using it in the top, and then using the transactionid for the currentstatus, and then the currentstatus as the transactionid.

**Comment by starsoccer** [comment]
On further testing the issue seems to be related to limit. As soon as I remove limit from my query, the query becomes correct

**Comment by starsoccer** [comment]
Issue still not resolved. Select and deleted are not effected, update causes the query to completely break

**Comment by dirkmc** [comment]
I'm seeing the same issue with select
```
knex('MyTable')
.limit(2)
.select('MyID', knex.raw(`MyCol = :someVal as MyCol`, { someVal: 'the val' }))
.toString()

"select top ('the val') [MyID], MyCol = 2 as MyCol from [MyTable]"
```

**Comment by starsoccer** [comment]
Hopefully someone can patch this bug

**Comment by dhensby** [comment]
fixed by #2066?

**Comment by elhigu** [comment]
yep https://runkit.com/embed/qh57udiyjwoi
